aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1221,6 +1222,33 @@ static void addmm_impl_cpu_(
   }
 }
 
+// TODO: This should work through the structured kernel approach of TORCH_IMPL_FUNC
+// TODO: ensure devices are handled
+Tensor addmm_zerotensor(const Tensor& self, const Tensor& mat1, const Tensor& mat2, const Scalar& beta, const Scalar& alpha){
+  TORCH_INTERNAL_ASSERT(self._is_zerotensor() || mat1._is_zerotensor() || mat2._is_zerotensor() || beta.equal(0) || alpha.equal(0));
+  const auto self_sizes = self.sizes();
+  Tensor result;
+  // beta*self + alpha*(mat1@mat2) = 0
+  if ((mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0)) &&
+      (self._is_zerotensor() || beta.equal(0))) {
+    result = at::_efficientzerotensor(self_sizes, self.options());
+    return result;
+  }
+  // beta*self
+  else if (mat1._is_zerotensor() || mat2._is_zerotensor() || alpha.equal(0)){
+    result.resize_as_(self);
+    result.copy_(self);
+    return beta.equal(1) ? result : result.mul_(beta);
+  }
+  // alpha*(mat1@mat2)
+  else if (self._is_zerotensor() || beta.equal(0)){
+    result = at::matmul(mat1, mat2);
+    return alpha.equal(1) ? result : result.mul_(alpha);
+  } else /* can't reach, see assert */{
+    return at::_efficientzerotensor(self_sizes, self.options());
```

Comment:
Without the `assert`, we will need to dispatch according to the appropriate device right?

Reply:
we can never reach here so we don't need this condition